app/models/embed/purl.rb

```diff
@@ -280,6 +277,32 @@ def video_duration_attr_str
           @file.xpath('./videoData').first.attributes['duration'].try(:text) if video_data?
         end
 
+        def video_duration
+          Duration.new(video_duration_attr_str) if video_duration_attr_str
+        rescue ISO8601::Errors::UnknownPattern
+          Honeybadger.notify(
+            "ResourceFile\#video_duration ISO8601::Errors::UnknownPattern: '#{video_duration_attr_str}'"
+          )
+          nil
+        end
+
+        def audio_data?
+          @file.xpath('./audioData').present?
+        end
+
+        def audio_duration_attr_str
+          @file.xpath('./audioData').first.attributes['duration'].try(:text) if audio_data?
+        end
+
+        def audio_duration
+          Duration.new(audio_duration_attr_str) if audio_duration_attr_str
```

Comment:
What would you think about having a `duration` method that calls `Duration.new` and passes in either the `audioData` or `videoData` value into it.  We might be able to reduce the number of methods that this class needs, which could reduce some complexity.

I think we're running into a lot of places where a lack of proper abstraction is making things clumsy, so totally understand if you don't want to do that as part of this work.  My hope is that we can eventually find the proper abstraction between `audioData`, `videoData`, and `imageData` (or at least start coalesce the approaches between video/audio).